seatunnel-e2e/seatunnel-connector-v2-e2e/connector-rabbitmq-e2e/src/test/java/org/apache/seatunnel/e2e/connector/rabbitmq/RabbitmqIT.java

```diff
@@ -0,0 +1,243 @@
+                 });
+            rows.add(row);
+        }
+        return Tuple2.apply(rowType, rows);
+    }
+
+    private void initRabbitMQ() {
+        try {
+            RabbitmqConfig config = new RabbitmqConfig();
+            config.setHost(rabbitmqContainer.getHost());
+            config.setPort(rabbitmqContainer.getFirstMappedPort());
+            config.setQueueName(QUEUE_NAME);
+            config.setVirtualHost("/");
+            config.setUsername(USERNAME);
+            config.setPassword(PASSWORD);
+            rabbitmqClient = new RabbitmqClient(config);
+        } catch (Exception e) {
+            throw new RuntimeException("init Rabbitmq error", e);
+        }
+    }
+
+    private RabbitmqClient initSinkRabbitMQ() {
+
+        try {
+            RabbitmqConfig config = new RabbitmqConfig();
+            config.setHost(rabbitmqContainer.getHost());
+            config.setPort(rabbitmqContainer.getFirstMappedPort());
+            config.setQueueName(SINK_QUEUE_NAME);
+            config.setVirtualHost("/");
+            config.setUsername(USERNAME);
+            config.setPassword(PASSWORD);
+            return new RabbitmqClient(config);
+        } catch (Exception e) {
+            throw new RuntimeException("init Rabbitmq error", e);
+        }
+    }
+
+    @AfterAll
+    @Override
+    public void tearDown() throws Exception {
+        if (connection != null) {
+            connection.close();
+        }
+        rabbitmqContainer.close();
+    }
+
+    @TestTemplate
+    public void testRabbitMQ(TestContainer container) throws Exception {
+        //send data to source queue before executeJob start in every testContainer
+        initSourceData();
+
+        //init consumer client before executeJob start in every testContainer
+        RabbitmqClient sinkRabbitmqClient =  initSinkRabbitMQ();
+
+        Set<String> resultSet = new HashSet<>();
+        Handover  handover = new Handover<>();
+        DefaultConsumer consumer = sinkRabbitmqClient.getQueueingConsumer(handover);
+        sinkRabbitmqClient.getChannel().basicConsume(SINK_QUEUE_NAME, true, consumer);
+        // assert execute Job code
+        Container.ExecResult execResult = container.executeJob("/rabbitmq-to-rabbitmq.conf");
+        Assertions.assertEquals(0, execResult.getExitCode());
+        //consume data when every  testContainer finished
+        //try to poll five times
+        for (int i = 0; i < 5; i++) {
+            Optional<Delivery> deliveryOptional = handover.pollNext();
+            if (deliveryOptional.isPresent()) {
+                Delivery delivery = deliveryOptional.get();
+                byte[] body = delivery.getBody();
+                resultSet.add(new String(body));
+            }
+        }
+        // close to prevent rabbitmq client consumer in the next TestContainer to consume
+        sinkRabbitmqClient.close();
+        //assert source and sink data
+        Assertions.assertTrue(resultSet.size() > 0);
```

Comment:
```suggestion
        Assertions.assertTrue(resultSet.size() == 5);
```

Reply:
> 

There are less than 5 pieces of data consumed on the actual souce side, because in the pollNext method of the RabbitmqSourceReader class, it is judged that after consuming data, the ` context signalNoMoreElement()`，this signal to the source that we have reached the end of the data. rabbitmq source connector on support streaming mode, this is for test。 Please give me some suggestions @hailin0